packages/views/src/components/pull-request/interfaces-store.ts

```diff
@@ -0,0 +1,105 @@
+import { ExecutionState } from '../execution/types'
+import {
+  RepoRepositoryOutput,
+  TypesListCommitResponse,
+  TypesPullReq,
+  TypesPullReqActivity,
+  TypesPullReqChecks,
+  TypesPullReqStats,
+  TypesRuleViolations
+} from './interfaces'
+
+export interface PullReqCount {
+  error: number
+  failure: number
+  pending: number
+  running: number
+  success: number
+  skipped: number
+  killed: number
+}
+
+export interface CheckInfo {
+  title: string
+  content: string
+  color: string
+  status: string
+}
+
+export interface CommentsInfoData {
+  header: string
+  content?: string
+  status: string
+}
+
+export interface RuleViolationData {
+  rule_violations: TypesRuleViolations[]
+}
+
+export interface RuleViolationArr {
+  data: RuleViolationData
+}
+
+export interface PullReqChecksDecisionProps {
+  overallStatus: ExecutionState | undefined
+  count: PullReqCount
+  error: unknown
+  data: TypesPullReqChecks | undefined
+  color: string
+  background: string
+  message: string
+  summaryText: string
+  checkInfo: CheckInfo
+}
+
+export interface PRPanelData {
+  conflictingFiles: string[] | undefined
+  requiresCommentApproval: boolean
+  atLeastOneReviewerRule: boolean
+  reqCodeOwnerApproval: boolean
+  minApproval: number
+  reqCodeOwnerLatestApproval: boolean
+  minReqLatestApproval: number
+  resolvedCommentArr?: { params: number[] }
+  PRStateLoading: boolean
+  ruleViolation: boolean
+  commentsLoading: boolean
+  commentsInfoData: CommentsInfoData
+  ruleViolationArr: RuleViolationArr | undefined
+}
+
+export interface PullRequestDataState {
+  repoMetadata: RepoRepositoryOutput | undefined
```

Comment:
1. we should avoid allowing `undefined` values in the store
2. if still needed, it's cleaner to make the value optional using `?`

```suggestion
  repoMetadata?: RepoRepositoryOutput
```

Reply:
will fix this

Reply:
will do point 2 first and then look into point